src/common/functions-v2/src/scalars/string_multi_args.rs

```diff
@@ -0,0 +1,308 @@
+                        }
+                    }
+                    ValueRef::Column(c) => {
+                        for idx in 0..size {
+                            for (arg_index, arg) in args.iter().skip(1).enumerate() {
+                                if arg_index != 0 {
+                                    unsafe {
+                                        builder.put_slice(c.index_unchecked(idx));
+                                    }
+                                }
+                                unsafe { builder.put_slice(arg.index_unchecked(idx)) }
+                            }
+                            builder.commit_row();
+                        }
+                    }
+                }
+
+                match len {
+                    Some(_) => Ok(Value::Column(Column::String(builder.build()))),
+                    _ => Ok(Value::Scalar(Scalar::String(builder.build_scalar()))),
+                }
+            }),
+        }))
+    });
+
+    // nullable concat ws
+    registry.register_function_factory("concat_ws", |_, args_type| {
+        if args_type.len() < 2 {
+            return None;
+        }
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "concat_ws",
+                args_type: vec![DataType::Nullable(Box::new(DataType::String)); args_type.len()],
+                return_type: DataType::Nullable(Box::new(DataType::String)),
+                property: FunctionProperty::default(),
+            },
+            calc_domain: Box::new(|_, _| None),
+            eval: Box::new(|args, _generics| {
+                type T = NullableType<StringType>;
+                let len = args.iter().find_map(|arg| match arg {
+                    ValueRef::Column(col) => Some(col.len()),
+                    _ => None,
+                });
+
+                let size = len.unwrap_or(1);
+                let new_args = args
+                    .iter()
+                    .map(|arg| arg.try_downcast::<T>().unwrap())
+                    .collect::<Vec<_>>();
+
+                let mut nullable_builder = T::create_builder(size, &[]);
+                match &new_args[0] {
+                    ValueRef::Scalar(None) => {
+                        return Ok(Value::Scalar(T::upcast_scalar(None)));
+                    }
+                    ValueRef::Scalar(Some(v)) => {
+                        let builder = &mut nullable_builder.builder;
+                        nullable_builder.validity.extend_constant(size, true);
+
+                        for idx in 0..size {
+                            for (arg_index, arg) in new_args.iter().skip(1).enumerate() {
+                                unsafe {
+                                    match arg.index_unchecked(idx) {
+                                        Some(s) if arg_index != 0 => {
+                                            builder.put_slice(v);
+                                            builder.put_slice(s);
```

Comment:
why the first arg is skipped?

Reply:
`CONCAT_WS(seperator, column1, ...)`

Reply:
but you've skip(1), which means column1 will be ignored